test/test_elbo.jl

```diff
@@ -314,6 +316,30 @@ function test_num_allowed_sd()
 end
 
 
+function test_elbo_supports_dual_numbers()
+    _, ea, _ = gen_two_body_dataset()
+
+    # create elbo arguments of type Float64 that compute the gradient but not the hessian
+    ea0 = ElboArgs(ea.images, ea.vp, ea.patches, ea.active_sources, calculate_hessian=false)
+
+    # create elbo arguments of the Dual number type, that compute the gradient but not
+    # the hessian (it doesn't matter that the "perturbation" are all zero, it's just
+    # for testing the speed and verifying that it works)
+    P = length(ea0.vp[1])
+    T = ForwardDiff.Dual{P * ea0.S, Float64}
```

Comment:
This is a very bad idea. It's type unstable, but more importantly, it could lead to very high chunk sizes, which could kill performance. For example, in the stacktrace in the OP, it seems you are constructing `Dual{88,Float64}`. That's going to absolutely thrash the stack. If you need to take that many simultaneous partial derivatives, you need to use chunking (which ForwardDiff does automatically) or use reverse-mode AD.